Add output metric attributes to Sensu 2.0 checks reference

apaskulin opened the pull request:
<!--- Thanks for submitting a pull request! -->
<!--- If you haven't yet, please read the contributing guide at: -->
<!--- https://github.com/sensu/sensu-docs/blob/master/CONTRIBUTING.md -->
<!--- Provide a general summary of your changes in the Title above. -->

## Description
<!--- Describe your changes in detail -->
This PR adds two missing attributes to the Sensu 2.0 checks reference docs: `output_metric_format` and `output_metric_handlers`.

Revised screenshot:

![screen shot 2018-09-12 at 8 46 30 pm](https://user-images.githubusercontent.com/11339965/45468477-aa880700-b6d9-11e8-86ef-e69fc92fbea3.png)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here: "Closes #555" -->
Closes #522

apaskulin committed:
add output metric attributes

nikkictl commented on content/sensu-core/2.0/reference/checks.md:
```diff
@@ -312,6 +312,20 @@ description  | If the event is to be silenced.
+|output_metric_format    |      |
+-------------|------
+description  | The metric format generated by the check command. Sensu supports the following metric formats: <br>`nagios_perfdata` ([Nagios Performance Data][nagios]) <br>`graphite_plaintext` ([Graphite Plaintext Protocol][graphite]) <br>`influxdb_line` ([InfluxDB Line Protocol][influx]) <br>`opentsdb_line` ([OpenTSDB Data Specification][open])<br><br>When a check includes an `output_metric_form ...
+required     | false (true if `output_metric_handlers` is configured)
```
I don't think `true if output_metric_handlers is configured` is actually enforced at the source level, although it is a good guideline. For example, one _can_ set `output_metric_handlers` and not specify `output_metric_format` however, metrics would not actually be extracted. Let me know if that makes sense?

nikkictl commented on content/sensu-core/2.0/reference/checks.md:
```diff
@@ -312,6 +312,20 @@ description  | If the event is to be silenced.
+
+|output_metric_handlers    |      |
+-------------|------
+description  | The metric formats expected by the `handlers` specified in the check definition. Sensu supports the following metric handler formats: <br>`nagios_perfdata` ([Nagios Performance Data][nagios]) <br>`graphite_plaintext` ([Graphite Plaintext Protocol][graphite]) <br>`influxdb_line` ([InfluxDB Line Protocol][influx]) <br>`opentsdb_line` ([OpenTSDB Data Specification][open])<br><br>When  ...
```
I think the description for the 4 metric formats is sufficient in `output_metric_formats`. `output_metric_handlers` are actually not confined to just those metric formats, since a metric handler can send a Sensu metric _anywhere_. For example, I believe Prometheus has a time series database that a handler could send these metrics to even though its not a supported line format for metric extraction. `output_metric_handlers` are simply a list of [Sensu handlers](https://github.com/sensu/sensu-docs/blob/master/content/sensu-core/2.0/reference/handlers.md) specifically designed to handle event data containing metrics.

apaskulin replied:
Interesting! So if you're looking to send metrics to a handler, would you use `output_metric_handlers` in place of regular `handlers`?

nikkictl replied:
Yes thats the intention! [Pipelined treats all handlers the same](https://github.com/sensu/sensu-go/blob/91df85824a5f521798e24d9ed6746dfd9e04c93c/backend/pipelined/handle.go#L47-L53) regardless though, but it's definitely best practice to reserve `output_metric_handlers` for handlers related to metrics.

nikkictl reviewed (changes requested):
Awesome work 👍! I just have a few comments regarding how things are happening at the source level.

apaskulin committed:
correct metric handler description

apaskulin committed:
add shortcut to event metrics

apaskulin committed:
add example

apaskulin commented:
Thanks, @nikkiki! I corrected the descriptions in https://github.com/sensu/sensu-docs/pull/730/commits/721db6be72e1257148e2ab621d79375cd8418728 and added an example of a metric check in https://github.com/sensu/sensu-docs/pull/730/commits/97244a73e28fab5c98bfdd0cb1b2ad02a74b9eea

nikkictl reviewed (approved):
Excellent job! Thanks for addressing those, and of course I always love a shoutout to my repo 😉 

![giphy](https://user-images.githubusercontent.com/10634078/45501561-60813e80-b736-11e8-9185-e731476274c9.gif)

apaskulin committed:
Merge branch 'master' into checks1
